### DISCUSSION TITLE
[QUESTION] Run migrations in conftest - fastapi

### DISCUSSION THREAD

**Discussion by jonatasoli** [sqlalchemy/alembic#871] [Usage Questions] [answered]
Hi,

I'm trying to make a project in fastapi and I would like to run all migrations in an asynchronous function in my conftest, is there any way to do this? Or would there be some way to build a fixture sync that does this job?

https://github.com/jonatasoli/fastapi-design-api-example/blob/main/app/conftest.py

I try

```
@pytest.fixture(scope="session", autouse=True)
async def apply_migrations() -> None:
     alembic.config.main(argv=["upgrade", "head"])
     alembic.config.main(argv=["downgrade", "base"])
```

**Comment by zzzeek** [comment, marked as answer]
see #870 for background on this issue.

short answer, running migrations using asyncio is unnecessary and it's much easier to use a normal DBAPI driver for your migrations.

**Reply by jonatasoli** [reply]
So I took a look, actually for me it would work if I could take a drop_all() and a create_all() syn form using a pytest fixture. So the best way would be in the contest I mount a synchronous engine and run the upgrade and downgrade?

**Reply by CaselIT** [reply]
for `drop_all()` and a `create_all()` you can use run_sync like in the example here https://docs.sqlalchemy.org/en/14/orm/extensions/asyncio.html#synopsis-core

**Reply by jonatasoli** [reply]
Thanks @CaselIT and @zzzeek 

I change my conftest with sync_run and found, thanks for support.

https://github.com/jonatasoli/fastapi-design-api-example/blob/main/app/conftest.py

**Comment by zzzeek** [comment]
oh oh sorry, I also forgot we have an async template as an example, which illustrates how to use run_sync:

https://github.com/sqlalchemy/alembic/blob/master/alembic/templates/async/env.py
